doc/src/parts/beginners_guide.rst

```diff
@@ -0,0 +1,482 @@
+contours can be shown by using the ``-c`` flag. In the *Python* example a ratio between the data
+and model function is shown below the plot. This can be done by appending the ``-r`` flag to
+*kafe2go*.
+
+.. bootstrap_collapsible::
+    :control_type: link
+    :control_text: non_linear_fit.yml
+
+    .. literalinclude:: ../../../examples/004_non_linear_fit/non_linear_fit.yml
+
+The dataset used to show that big uncertainties on the x axis can cause the fit to be non linear
+follows here.
+Keep in mind, that *kafe2go* will perform a line fit if no fit function has been specified.
+In order do add a grid to the contours, run *kafe2go* with the ``--grid all`` flag.
+So to plot with asymmetric errors, the profiles and contour as well as a grid run
+``kafe2go x_errors.yml -a -c --grid all``
+
+.. bootstrap_collapsible::
+    :control_type: link
+    :control_text: x_errors.yml
+
+    .. literalinclude:: ../../../examples/004_non_linear_fit/x_errors.yml
+
+Python
+------
+
+The according lines to display asymmetric uncertainties and to create the contour plot are
+highlighted in the code example below.
+
+.. bootstrap_collapsible::
+    :control_type: link
+    :control_text: non_linear_fit.py
+
+    .. literalinclude:: ../../../examples/004_non_linear_fit/non_linear_fit.py
+        :lines: 27-
+        :emphasize-lines: 23, 27, 29-31
+
+The example to show that uncertainties on the x axis can cause a non linear fit uses the *YAML*
+dataset given in the *kafe2go* section.
+
+.. bootstrap_collapsible::
+    :control_type: link
+    :control_text: non_linear_fit.py
+
+    .. literalinclude:: ../../../examples/004_non_linear_fit/x_errors.py
+        :lines: 19-
+
+
+Example 5: Histogram Fit
+========================
+
+*kafe2* is not only capable of performing XY-Fits. This example shows how to handle a one
+dimensional dataset.
+The distribution of a random stochastic variable follows a probability density function.
+This density needs to be normalized.
+The fit will determine the parameters of that density function, which the dataset is most likely to
+follow.
+If the given function is not normalized, please make sure to add an independent scale factor.
+The hist fit will try to normalize the function to one for the bin range.
+To get to the height of a bin, please multiply the results of the fitted function with the amount
+of entries N of the histogram.
+
+.. figure:: ../_static/img/005_hist_fit.png
+    :alt: A histogram fit performed with kafe2.
+
+kafe2go
+-------
+
+*kafe2go* currently requires extra formatting to perform a histogram fit. This will be changed in a
+future update. At the moment it's simpler to creat a *Python* script.
+
+Python
+------
+
+To use a histogram fit in a *Python* script, just import it with
+:python:`from kafe2 import HistContainer, HistFit`.
+
+The creation a a histogram requires the user to set the limits of the histogram and the amount of
+bins.
+Alternatively the bin edges for each bin can be set.
```

Comment:
Alternatively the bin edges for each bin can be set. -> Alternatively the bin edges for each bin can be set manually.